### PR TITLE
Add migrations to DP EntityFrameworkCoreSample

### PR DESCRIPTION
**Pull request by guardrex**
Summary of the changes:
 - Add migrations to Data Protection EntityFrameworkCoreSample

Per discussion: https://github.com/aspnet/Docs/issues/9864#issuecomment-446785281

```csharp
.HasAnnotation("ProductVersion", "2.2.0-rtm-35687")
```

... seems unusual. Should it just be "2.2.0"? If so, I'll (manually) change those two spots on the next commit. Something similar happens with my aspnet-codegenerator tooling (https://github.com/dotnet/cli/issues/10459).

cc: @natemcmaster

### CODE REVIEW

File: src/DataProtection/samples/EntityFrameworkCoreSample/Migrations/20181219214311_AddDataProtectionKeys.Designer.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+{
+    [DbContext(typeof(DataProtectionKeyContext))]
+    [Migration("20181219214311_AddDataProtectionKeys")]
+    partial class AddDataProtectionKeys
+    {
+        protected override void BuildTargetModel(ModelBuilder modelBuilder)
+        {
+#pragma warning disable 612, 618
+            modelBuilder
+                .HasAnnotation("ProductVersion", "2.2.0-rtm-35687")
```
Comment (natemcmaster) [inline src/DataProtection/samples/EntityFrameworkCoreSample/Migrations/20181219214311_AddDataProtectionKeys.Designer.cs]: This seems odd. Samples in our master branch should be using EFCore 3.0. Which version of the dotnet CLI did you use to generate this migration?

Comment (guardrex) [reply src/DataProtection/samples/EntityFrameworkCoreSample/Migrations/20181219214311_AddDataProtectionKeys.Designer.cs]: 2.2.100 ... but I had the preview on here earlier.

Comment (natemcmaster) [reply src/DataProtection/samples/EntityFrameworkCoreSample/Migrations/20181219214311_AddDataProtectionKeys.Designer.cs]: Ok. I'm fine as is. This is "just a sample" after all, and as long as EFCore doesn't break the migrations API in 3.0, there's no real concern here.

**Comment by natemcmaster** [comment]
Hmm, looks like there are compiler errors.

https://dnceng.visualstudio.com/public/_build/results?buildId=61337&view=logs&jobId=f31c9f97-4411-58e7-49ac-fc73f645e6b6&taskId=2bcaa12b-2f4b-5b1f-c519-10308f653190&lineStart=2089&lineEnd=2101&colStart=1&colEnd=46
```
2018-12-19T22:31:37.0571934Z D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\Migrations\20181219214311_AddDataProtectionKeys.cs(2,37): error CS0234: The type or namespace name 'Migrations' does not exist in the namespace 'Microsoft.EntityFrameworkCore' (are you missing an assembly reference?) [D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\EntityFrameworkCoreSample.csproj]
2018-12-19T22:31:37.0572603Z D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\Migrations\20181219214311_AddDataProtectionKeys.Designer.cs(5,37): error CS0234: The type or namespace name 'Migrations' does not exist in the namespace 'Microsoft.EntityFrameworkCore' (are you missing an assembly reference?) [D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\EntityFrameworkCoreSample.csproj]
2018-12-19T22:31:37.0573230Z D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\Migrations\20181219214311_AddDataProtectionKeys.cs(6,50): error CS0246: The type or namespace name 'Migration' could not be found (are you missing a using directive or an assembly reference?) [D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\EntityFrameworkCoreSample.csproj]
2018-12-19T22:31:37.0573894Z D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\Migrations\20181219214311_AddDataProtectionKeys.Designer.cs(12,6): error CS0246: The type or namespace name 'MigrationAttribute' could not be found (are you missing a using directive or an assembly reference?) [D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\EntityFrameworkCoreSample.csproj]
2018-12-19T22:31:37.0574619Z D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\Migrations\20181219214311_AddDataProtectionKeys.Designer.cs(12,6): error CS0246: The type or namespace name 'Migration' could not be found (are you missing a using directive or an assembly reference?) [D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\EntityFrameworkCoreSample.csproj]
2018-12-19T22:31:37.0575832Z D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\Migrations\20181219214311_AddDataProtectionKeys.cs(8,36): error CS0246: The type or namespace name 'MigrationBuilder' could not be found (are you missing a using directive or an assembly reference?) [D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\EntityFrameworkCoreSample.csproj]
2018-12-19T22:31:37.0576354Z D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\Migrations\20181219214311_AddDataProtectionKeys.cs(25,38): error CS0246: The type or namespace name 'MigrationBuilder' could not be found (are you missing a using directive or an assembly reference?) [D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\EntityFrameworkCoreSample.csproj]
2018-12-19T22:31:37.0576957Z D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\Migrations\20181219214311_AddDataProtectionKeys.cs(8,33): error CS0115: 'AddDataProtectionKeys.Up(MigrationBuilder)': no suitable method found to override [D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\EntityFrameworkCoreSample.csproj]
2018-12-19T22:31:37.0577603Z D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\Migrations\20181219214311_AddDataProtectionKeys.cs(25,33): error CS0115: 'AddDataProtectionKeys.Down(MigrationBuilder)': no suitable method found to override [D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\EntityFrameworkCoreSample.csproj]
2018-12-19T22:31:37.0581494Z D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\Migrations\20181219214311_AddDataProtectionKeys.Designer.cs(15,33): error CS0115: 'AddDataProtectionKeys.BuildTargetModel(ModelBuilder)': no suitable method found to override [D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\EntityFrameworkCoreSample.csproj]
2018-12-19T22:31:37.0582205Z D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\Migrations\DataProtectionKeyContextModelSnapshot.cs(11,59): error CS0246: The type or namespace name 'ModelSnapshot' could not be found (are you missing a using directive or an assembly reference?) [D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\EntityFrameworkCoreSample.csproj]
2018-12-19T22:31:37.0582720Z D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\Migrations\DataProtectionKeyContextModelSnapshot.cs(13,33): error CS0115: 'DataProtectionKeyContextModelSnapshot.BuildModel(ModelBuilder)': no suitable method found to override [D:\a\1\s\src\DataProtection\samples\EntityFrameworkCoreSample\EntityFrameworkCoreSample.csproj]
2018-12-19T22:31:37.0583145Z     1 Warning(s)
2018-12-19T22:31:37.0583533Z     12 Error(s)
```

**Comment by natemcmaster** [comment]
I think the problem is that the sample doesn't reference a database provider.

https://github.com/guardrex/AspNetCore/blob/798b49ffbd900ba8e841ae6f54f5087e74322e83/src/DataProtection/samples/EntityFrameworkCoreSample/EntityFrameworkCoreSample.csproj#L8-L12

Are these migrations for SqlServer, Sqlite, Postgres, something else?

**Comment by guardrex** [comment]
SqlServer

**Comment by natemcmaster** [comment]
Can you add that to the project?

**Comment by guardrex** [comment]
I'll get this on Friday. I was buried in HTTP.sys topic updates today. 🏃🏃🏃🏃😅

**Comment by guardrex** [comment]
Idk how you handle the package versions. I took a *guess* from looking at the others ...

```
Version="$(MicrosoftEntityFrameworkCoreSqlServerPackageVersion)"
```

I also removed the "-rtm-X" from the `Product Version` values. However, I'm on 2.2 here, the migrations are still calling for "2.2," and this is built for 3.0-preview. Idk how that situation is handled in the engineering repos.

**Review by natemcmaster — APPROVED** [review]

**Comment by natemcmaster** [comment]
Thanks @guardrex !

**Pull request merged**